src/quick-lint-js/parse.h

```diff
@@ -3006,6 +3006,19 @@ class parser {
           break;
         }
 
+        case token_type::kw_new: {
+          const char8 *here = this->lexer_.end_of_previous_token();
+          this->error_reporter_->report(error_missing_equal_after_variable{
+              .expected_equal = source_code_span(here, here),
+          });
+          expression *ast = this->parse_expression_remainder(
+              variable,
+              precedence{.commas = false, .in_operator = allow_in_operator});
+          this->lexer_.insert_semicolon();
```

Comment:
Hmm, inserting a semicolon after the expression unconditionally seems wrong. What about the following code?
```js
let a new Array(), b = a.length;
```
I think your change will cause an error at `,` (in addition to the expected error between `a` and `new`).